addons/point_of_sale/models/pos_session.py

```diff
@@ -278,15 +282,468 @@ def _validate_session(self):
+                    else:
+                        split_receivables[payment] = self._update_amounts(split_receivables[payment], {'amount': amount}, date)
                 else:
-                    return self._warning_balance_closing()
+                    key = payment.payment_method_id
+                    if payment.payment_method_id.is_cash_count:
+                        combine_receivables_cash[key] = self._update_amounts( combine_receivables_cash[key], {'amount': amount}, date)
+                    else:
+                        combine_receivables[key] = self._update_amounts(combine_receivables[key], {'amount': amount}, date)
+
+            if order.is_invoiced:
+                # Combine invoice receivable lines
+                key = order.partner_id.property_account_receivable_id.id
+                invoice_receivables[key] = self._update_amounts(invoice_receivables[key], {'amount': order.amount_total}, order.date_order)
+                # side loop to gather receivable lines by account for reconciliation
+                for move_line in order.account_move.line_ids.filtered(lambda aml: aml.account_id.internal_type == 'receivable'):
+                    order_account_move_receivable_lines[move_line.account_id.id] |= move_line
             else:
-                self._validate_session()
+                for line in map(self._prepare_line, order.lines):
+                    # Combine sales/refund lines
+                    sale_key = (
+                        # account
+                        line['income_account_id'],
+                        # sign
+                        -1 if line['amount'] < 0 else 1,
+                        # for taxes
+                        tuple((tax['id'], tax['account_id'], tax['tax_repartition_line_id']) for tax in line['taxes']),
+                    )
+                    sales[sale_key] = self._update_amounts(sales[sale_key], {'amount': line['amount']}, line['date_order'])
+                    # Combine tax lines
+                    tax_keys = set()  # track calculated tax which will be rounded every after order
+                    for tax in line['taxes']:
+                        tax_key = (tax['account_id'], tax['tax_repartition_line_id'], tax['id'], tuple(tax['tag_ids']))
+                        tax_keys.add(tax_key)
+                        taxes[tax_key] = self._update_amounts(
+                            taxes[tax_key],
+                            {'amount': tax['amount'], 'base_amount': tax['base']},
+                            tax['date_order'],
+                            round=not rounded_globally
+                        )
+                if rounded_globally:
+                    for tax_key in tax_keys:
+                        taxes[tax_key] = self._round_amounts(taxes[tax_key])
+
+                # Combine stock lines
+                stock_moves = self.env['stock.move'].search([
```

Comment:
you can put all stock moves stuff under:

if self.company_id.anglo_saxon_accounting